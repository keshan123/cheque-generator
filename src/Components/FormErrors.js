import React from "react";
import PropTypes from "prop-types";

const FormErrors = ({ formErrors }) => (
  <div className="form-input__error-message">
    {Object.keys(formErrors).map((fieldName, index) => {
      if (formErrors[fieldName].length > 0) {
        return (
          <p className="form-input__error-message__individual" key={index}>
            {fieldName} {formErrors[fieldName]}
          </p>
        );
      } else {
        return "";
      }
    })}
  </div>
);

FormErrors.propTypes = {
  formErrors: PropTypes.shape({
    payee: PropTypes.string,
    amount: PropTypes.string,
    date: PropTypes.string
  })
};

export default FormErrors;
