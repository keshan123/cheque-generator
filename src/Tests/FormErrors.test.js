import React from "react";
import FormErrors from "../Components/FormErrors";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const formErrors = {
    payee: "mockpayee error",
    amount: "mockamount error",
    date: "mockdate error"
  };
  const tree = renderer
    .create(<FormErrors formErrors={formErrors}></FormErrors>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
