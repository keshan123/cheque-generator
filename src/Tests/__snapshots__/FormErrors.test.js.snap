// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-input__error-message"
>
  <p
    className="form-input__error-message__individual"
  >
    payee
     
    mockpayee error
  </p>
  <p
    className="form-input__error-message__individual"
  >
    amount
     
    mockamount error
  </p>
  <p
    className="form-input__error-message__individual"
  >
    date
     
    mockdate error
  </p>
</div>
`;
